receiver/aerospikereceiver/scraper.go

```diff
@@ -0,0 +1,217 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aerospikereceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/aerospikereceiver"
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strconv"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/aerospikereceiver/internal/metadata"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/aerospikereceiver/internal/model"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/receiver/scrapererror"
+	"go.uber.org/zap"
+)
+
+// aerospikeReceiver is a metrics receiver using the Aerospike interface to collect
+type aerospikeReceiver struct {
+	params   component.ReceiverCreateSettings
+	config   *Config
+	consumer consumer.Metrics
+	host     string // host/IP of configured Aerospike node
+	port     int    // port of configured Aerospike node
+	mb       *metadata.MetricsBuilder
+	logger   *zap.Logger
+}
+
+// newAerospikeReceiver creates a new aerospikeReceiver connected to the endpoint provided in cfg
+//
+// If the host or port can't be parsed from endpoint, an error is returned.
+func newAerospikeReceiver(params component.ReceiverCreateSettings, cfg *Config, consumer consumer.Metrics) (*aerospikeReceiver, error) {
+	host, portStr, err := net.SplitHostPort(cfg.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("endpoint: %w", err)
+	}
+
+	port, err := strconv.ParseInt(portStr, 10, 32)
+	if err != nil {
+		return nil, fmt.Errorf("port: %w", err)
+	}
+
+	return &aerospikeReceiver{
+		params:   params,
+		config:   cfg,
+		consumer: consumer,
+		host:     host,
+		port:     int(port),
+		mb:       metadata.NewMetricsBuilder(cfg.Metrics),
+		logger:   params.Logger.Named("aerospikereceiver"),
+	}, nil
+}
+
+// scrape scrapes both Node and Namespace metrics from the provided Aerospike node.
+// If CollectClusterMetrics is true, it then scrapes every discovered node
+func (r *aerospikeReceiver) scrape(ctx context.Context) (pmetric.Metrics, error) {
+	errs := &scrapererror.ScrapeErrors{}
+	now := pcommon.NewTimestampFromTime(time.Now().UTC())
+	client, err := newASClient(r.host, r.port, r.config.Timeout)
+	if err != nil {
+		r.logger.Warn(fmt.Sprintf("failed to connect: %s", err.Error()))
+		return r.mb.Emit(), nil
+	}
+	defer client.Close()
+
+	info, err := client.Info()
+	if err != nil {
+		return r.mb.Emit(), err
+	}
+	r.emitNode(info, client, now)
+
+	if r.config.CollectClusterMetrics {
+		r.logger.Sugar().Debugf("Collecting %d peer nodes", len(info.Services))
```

Comment:
I don't have any strong feelings against using sugared logging, but it isn't something I've really seen used in other components, I've only ever seen the field based logging used. Might make more sense to just follow convention there.